nixos/doc/manual/administration/service-mgmt.xml

```diff
@@ -62,11 +69,63 @@ Jan 07 15:55:57 hagbard systemd[1]: Started PostgreSQL Server.
 # systemctl start postgresql.service
 # systemctl restart postgresql.service
 </screen>
-  These operations are synchronous: they wait until the service has finished
-  starting or stopping (or has failed). Starting a unit will cause the
-  dependencies of that unit to be started as well (if necessary).
- </para>
-<!-- - cgroups: each service and user session is a cgroup
+   These operations are synchronous: they wait until the service has finished
+   starting or stopping (or has failed). Starting a unit will cause the
+   dependencies of that unit to be started as well (if necessary).
+  </para>
+  <!-- TODO: document cgroups, draft:
+   each service and user session is a cgroup
 
-- cgroup resource management -->
+   - cgroup resource management -->
+ </section>
+ <section xml:id="sect-nixos-systemd-nixos">
+  <title>Systemd in NixOS</title>
+  <para>
+   Packages in Nixpkgs sometimes provide Systemd units with them, usually in
+   e.g <literal>#pkg-out#/lib/systemd/</literal>. Putting such a package in
+   <literal>environment.systemPackages</literal> doesn't make the service
+   available to users or the system. Usually system services are defined using
+   NixOS modules written by the community. These services are defined in Nixpkgs'
+   <link xlink:href="https://github.com/NixOS/nixpkgs/tree/master/nixos/modules">
+   <literal>nixos/modules/</literal> directory </link>. We prefer to not rely on
+   upstream provided Systemd services, since this gives us more control over how
+   the service behaves. Upstream's provided Systemd services are taken into
+   consideration though when modules are written.
+  </para>
+  <para>
+   User Systemd services on the other hand, can be treated differently. Given a package
+   that has a Systemd file at <literal>#pkg-out#/lib/systemd/user/</literal>, you
+   don't have to write a NixOS module, or configure a
+   <link linkend="opt-systemd.user.services">
+   <literal>systemd.user.services</literal></link>. You can imperatively enable it
+   by adding the package's attribute to <link linkend="opt-environment.systemPackages">
+   <literal>systemd.packages</literal></link> and then do this (e.g):
+<screen>
+<prompt>$ </prompt>mkdir -p ~/.config/systemd/user/default.target.wants
```

Comment:
Do you really have to create that dir manually or isn't that done by the `systemctl --user enable …` below?

Reply:
It may not exist if it's a first time when you use systemd for anything. The point of this `<screen>`, is to enable a Systemd user service **without** `systemctl --user enable`.

Reply:
From experience, I can confirm this is the correct way to enable user services. 
Using `systemctl --user enable` will create a link that will break once the package hash changes and the directory may not exist.